### DISCUSSION TITLE
MD22 MD17 train and test data split

### DISCUSSION THREAD

**Discussion by jhmlam** [ACEsuit/mace#624] [Q&A] [answered]
Hi @gabor1 @davkovacs @ilyes319 ,

Thank you for sharing, developing and maintaining the dataset and mace.
I saw on `https://mace-docs.readthedocs.io/en/latest/examples/training_examples.html` that there is a `nanotube_test.xyz` file. However, when I go to `http://www.sgdml.org/#datasets`, it seems that there is no official train test split for MD 17 and 22, though there are splitted xyz for the ccsd datasets there. 

Is it possible to provide a simple script to do the split producing xyz similar to 

https://github.com/ilyes319/mace-tutorials/blame/1172df9fb93cbaad6b3abda87526ec3484ef33fe/mace-users/MACE_users.ipynb#L469

such that it roughly matches `Table I` in the report `Kovács DP, Batatia I, Arany ES, Csányi G. Evaluation of the MACE force field architecture: From medicinal chemistry to materials science. J Chem Phys. 2023 Jul 28;159(4):044118. doi: 10.1063/5.0155322. PMID: 37522405.` 

Thank you!

Best,
Jordy

**Comment by jhmlam** [comment]
Sorry for doubling. Indeed, which files "nanotube_test.xyz", "nanotube_large.xyz" are referring to on http://www.sgdml.org/#datasets. These files were not inside the `Double-walled nanotube` folder. So I cannot actually run the example exactly as written.


```
python <mace_repo_dir>/mace/cli/run_train.py \
    --name="nanotube_large_r55" \
    --train_file="nanotube_large.xyz" \
    --valid_fraction=0.05 \
    --test_file="nanotube_test.xyz" \
    --E0s="average" \
    --model="MACE" \
    --num_interactions=2 \
    --num_channels=256 \
    --max_L=2 \
    --correlation=3 \
    --r_max=5.0 \
    --forces_weight=1000 \
    --energy_weight=10 \
    --batch_size=2 \
    --valid_batch_size=2 \
    --max_num_epochs=650 \
    --start_swa=450 \
    --scheduler_patience=5 \
    --patience=15 \
    --eval_interval=3 \
    --ema \
    --swa \
    --swa_forces_weight=10 \
    --error_table='PerAtomMAE' \
    --default_dtype="float64"\
    --device=cuda \
    --seed=123 \
    --restart_latest \
    --save_cpu

```

**Comment by ilyes319** [comment, marked as answer]
Hello,

You can just run the following script that will select randomly 5% of the set for validation and 95% for training as detailed in the appendix of the paper. Make sure to convert the units from kcal/mol to eV before training for both energies and forces.

```
python <mace_repo_dir>/mace/cli/run_train.py \
    --name="nanotube_large_r55" \
    --train_file="md22_double-walled_nanotube.xyz" \
    --valid_fraction=0.05 \
    --test_file="md22_double-walled_nanotube.xyz" \
    --E0s="average" \
    --model="MACE" \
    --num_interactions=2 \
    --num_channels=256 \
    --max_L=2 \
    --correlation=3 \
    --r_max=5.0 \
    --forces_weight=1000 \
    --energy_weight=10 \
    --batch_size=2 \
    --valid_batch_size=2 \
    --max_num_epochs=650 \
    --start_swa=450 \
    --scheduler_patience=5 \
    --patience=15 \
    --eval_interval=3 \
    --ema \
    --swa \
    --swa_forces_weight=10 \
    --error_table='PerAtomMAE' \
    --default_dtype="float64"\
    --device=cuda \
    --seed=123 \
    --restart_latest \
    --save_cpu
```

**Reply by jhmlam** [reply]
Thank you @ilyes319 (also for the important remark on eV). Now it works. The flags 
```
        --energy_key="Energy" \
        --forces_key="forces" \ 
```
were also used, otherwise it gives zero loss.

**Reply by stefanozampini** [reply]
I'm also trying to run the training examples. @ilyes319 What do I need to do to "Make sure to convert the units from kcal/mol to eV before training for both energies and forces." ? Is there a tutorial/example I can take a look at?

**Reply by gabor1** [reply]
Have a look at your training data and determine the units being used. The original publication uses different units than what we use for mace.

**Reply by stefanozampini** [reply]
@gabor1 thanks for the quick reply. I'm not at all an expert MD, nor do I know where or how to modify the training script to change the units; my intent is to study the approximation properties of second-order methods like Gauss-Newton for the training problem; see, e.g., https://arxiv.org/abs/2403.12188, reproducing published results as much as possible in terms of datasets. The nanotube test is the first one I attempted to reproduce, and I found that the exact datasets are not available, as explained in this discussion. I'm not specifically interested in the PyTorch backend; I can switch to JAX. Do you have suggestions for other experiments I can reproduce?

**Reply by gabor1** [reply]
You don't modify training scripts. You ascertain that when you use a dataset which says the energy of this molecule is X, what are the units of X. If it's not what you want, you convert the units in your data file.

Different ppl use different units. Science is messy. Mace uses eV for energies, so all data files you feed it need to have energies in eV (and forces in eV/Å and positions in Å). 

You are lucky it's 2026, if you are unsure what a datafile's units are, I bet leading LLMs will tell you.
